x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -392,6 +395,30 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
+  async isEditable(): Promise<boolean> {
+    if (!getMapAppConfig().enableDrawingFeature) {
+      this._isEditable = false;
+      return this._isEditable;
+    }
+    if (this._isEditable === undefined) {
+      if (!this.indexPattern) {
```

Comment:
this.getIndexPattern already checks for cached "this.indexPattern" and returns that if available. You can just use "const indexPattern = await this.getIndexPattern()" instead of wrapping getIndexPattern in this if block. This will be consistent with how getIndexPattern is used everywhere else in the file.